upgrade/php/ps_910_migrate_cart_rule_types.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+use PrestaShop\Module\AutoUpgrade\Database\DbWrapper;
+
+/**
+ * Migrate cart_rule.type to the new cart_rule_type structure
+ * This function:
+ * 1. Extracts unique types from cart_rule table
+ * 2. Populates cart_rule_type table
+ * 3. Adds id_cart_rule_type column to cart_rule
+ * 4. Migrates data from type to id_cart_rule_type
+ * 5. Drops the old type column
+ *
+ * @return void
+ *
+ * @throws \PrestaShop\Module\AutoUpgrade\Exceptions\UpdateDatabaseException
+ */
+function ps_910_migrate_cart_rule_types()
+{
+    $now = date('Y-m-d H:i:s');
+
+    // Check if cart_rule table has the old 'type' column
+    $columns = DbWrapper::executeS('SHOW COLUMNS FROM `' . _DB_PREFIX_ . 'cart_rule` WHERE Field = "type"');
+
+    if (empty($columns)) {
+        return;
+    }
+
+    // Get all discount types from cart_rule table
+    $existingTypes = DbWrapper::executeS(
+        'SELECT DISTINCT `type` FROM `' . _DB_PREFIX_ . 'cart_rule` WHERE `type` IS NOT NULL AND `type` != ""'
+    );
+
+    $languages = DbWrapper::executeS('SELECT `id_lang` FROM `' . _DB_PREFIX_ . 'lang`');
+
+    foreach ($existingTypes as $typeRow) {
```

Comment:
I don't think this one has much value, it populates the cart rule types based on the ones in the DB but this column will always be empty after a first upgrade

Reply:
Instead of populating the cart rule types based on the content from cart rule we should do the opposite: try to deduce from the cart rule configuration if it matches one of our types But I'm not sure this should be done by the upgrade module

It's more likely a feature on the core, yet to be developed, that will help migrate old cart rules to new discounts